synapse/event_auth.py

```diff
@@ -207,13 +208,20 @@ def too_big(field):
         too_big("event")
 
 
-def _can_federate(event, auth_events):
+def _can_federate(
+    event: EventBase, auth_events: Dict[Tuple[str, str], EventBase]
+) -> bool:
     creation_event = auth_events.get((EventTypes.Create, ""))
+    # There should always be a creation event, but if not don't federate.
+    if not creation_event:
+        return False
```

Comment:
This was necessary to make mypy happy, it shouldn't happen but seems like a good guard.